x-pack/plugin/esql/qa/testFixtures/src/main/resources/spatial-grid.csv-spec

```diff
@@ -0,0 +1,887 @@
+38         | POINT (-9.429514887252529 25.497624435045413)  | e
+34         | POINT (-111.8071846965262 52.464381378993174)  | c
+30         | POINT (28.7045472683385 -14.706001980230212)   | k
+28         | POINT (159.52750137208827 -25.555616633001982) | r
+22         | POINT (-4.410395708612421 54.90304926367985)   | g
+21         | POINT (-69.40534970590046 50.93379438189523)   | f
+17         | POINT (114.05526293222519 -10.898114638950895) | q
+16         | POINT (147.40052131412085 21.054660080408212)  | x
+13         | POINT (63.64716878519035 54.37333276101317)    | v
+12         | POINT (-39.53510569408536 -11.72166372067295)  | 7
+// end::st_geohash-grid-result[]
+;
+
+gridGeohashQuery
+required_capability: spatial_grid
+
+FROM airports
+| WHERE ST_GEOHASH(location, 1) == ST_GEOHASH_TO_LONG("7")
+| EVAL geohash = ST_GEOHASH(location, 2)
+| STATS
+    count = COUNT(*),
+    centroid = ST_CENTROID_AGG(location)
+      BY geohash
+| EVAL geohashString = ST_GEOHASH_TO_STRING(geohash)
+| KEEP count, centroid, geohashString
+| SORT count DESC, geohashString ASC
+;
+
+count:long | centroid:geo_point                             | geohashString:keyword
+4          | POINT (-35.22712387377396 -7.672965489327908)  | 7n
+3          | POINT (-41.86607404612005 -3.8085224060341716) | 7p
+2          | POINT (-42.66381660941988 -22.868987743277103) | 75
+2          | POINT (-42.12605922482908 -19.945099228061736) | 7h
+1          | POINT (-38.3347990270704 -12.91436152998358)   | 7j
+;
+
+gridGeohashStatsByBounds
+required_capability: spatial_grid
+
+FROM airports
+| EVAL geohash = ST_GEOHASH(location, 2, TO_GEOSHAPE("POLYGON((0.0 30.0, 12.0 30.0, 12.0 60.0, 0.0 60.0, 0.0 30.0))"))
+| WHERE geohash IS NOT NULL
+| STATS
+    count = COUNT(*),
+    centroid = ST_CENTROID_AGG(location)
+      BY geohash
+| EVAL geohashString = ST_GEOHASH_TO_STRING(geohash)
+| KEEP count, centroid, geohashString
+| SORT count DESC, geohashString ASC
+;
+
+count:long | centroid:geo_point                            | geohashString:keyword
+19         | POINT (6.360728044651057 47.94084087577894)   | u0
+10         | POINT (15.350638423115015 47.80751353036612)  | u2
+9          | POINT (18.5217544157058 42.1394603792578)     | sr
+8          | POINT (6.351574736181647 51.8981519783847)    | u1
+7          | POINT (5.268637698941997 42.747250193330856)  | sp
+7          | POINT (17.092350951528974 53.365471504096476) | u3
+5          | POINT (16.2651440910995 58.812188878655434)   | u6
+4          | POINT (7.7012718468904495 36.39783004182391)  | sn
+3          | POINT (14.222751930356026 37.168446206487715) | sq
+3          | POINT (7.318722177296877 59.788265260867774)  | u4
+2          | POINT (16.706149326637387 32.37822346854955)  | sm
+;
+
+gridGeohashStatsByPointsBounds
+required_capability: spatial_grid
+
+FROM airports
+| EVAL points = ["POINT(0.0 30.0)", "POINT(12.0 60.0)"]
+| EVAL geohash = ST_GEOHASH(location, 2, TO_GEOPOINT(points))
+| WHERE geohash IS NOT NULL
```

Comment:
Why is this needed?

Reply:
I think I understand now. It is strange how the bounds are defined here.

Reply:
Support for bounds defined by points is now removed.